### DISCUSSION TITLE
When create pull request default target branch to the parent branch that the current branch was branched from

### DISCUSSION THREAD

**Discussion by Landmark-Adrian** [community/community#13268] [Pull Requests]
In our team we create a feature branch from develop for our JIRA task, when we have committed our changes and raise a pull request it always defaults to merging back into 'main' instead of back into the parent branch which should be 'develop'.

So we ALWAYS end up changing the target branch back to 'develop' - and sometimes we forget (but usually someone on the PR notices notices) - it would be BAD if it did merge to 'main' because we have deployment processes running off of merging to 'main'.

**Comment by heindel** [comment]
for this case, I think, the default branch configuration would do the trick

https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-branches-in-your-repository/changing-the-default-branch

**Reply by MarkoJAho** [reply]
Changing the default is ok, if one only has one release branch to maintain. However, there could be additional development on many existing release branches, and similarly bugfixes to those release branches which are under support.

So in practice the UI should have by default set the original branch the feature/bugfix branch was created from. One can then override the default in the UI as needed if the merge is not intended to the suggested one.

Alternatively at least the PR page should have a much bigger font or other emphasis for the branch merge suggestion. Coming from Bitbucket there the merge target branch is hugely more prominently visible as the first action to be done when creating a new PR. It is very rare to end up with an incorrect merge on that tool, but this trips many developers moving to GitHub.

**Reply by MarkoJAho** [reply]
Actually if the Default branch could be set to be empty (or "not set") for the PR separately from the other places referring to the Default branch, it would then be able to prevent creating the PR without having to set the target branch. There would need to be a check that the target branch is something other than empty in the CreatePR button to be enabled. This way the page would force the user to always take conscious step where the PR is to be merged.
